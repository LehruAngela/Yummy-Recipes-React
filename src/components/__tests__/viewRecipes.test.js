import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import ViewRecipes from '../viewRecipes';

describe('ViewRecipes', () => {
    const params = {
        match: {
          params: {
            id: 1,
          },
        },
      };
    const wrapper = shallow(<ViewRecipes match={{ params }}/>);
    const preventDefault = jest.fn();

    it('renders properly', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
        }); 
    
    it('handles ViewRecipes', () => {
        expect(wrapper.instance().handleViewRecipe({ preventDefault }));
        });

    it('handles ViewRecipes', () => {
        expect(wrapper.instance().handleViewRecipe({ preventDefault }));
    });

    it('handles ViewRecipes', () => {
        expect(wrapper.instance().handleDeleteRecipe({ preventDefault }));
    });

    it('handles ViewRecipes', () => {
        expect(wrapper.instance().handleEditRecipe({ preventDefault }));
    });

    it('handles ViewRecipes', () => {
        expect(wrapper.instance().handlePageChange({ preventDefault }));
    });


})