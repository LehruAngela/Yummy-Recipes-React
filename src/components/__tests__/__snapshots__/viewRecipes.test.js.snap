// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewRecipes renders properly 1`] = `
<div>
  <div
    className="header"
  >
    <Component />
    <div>
      <h2
        className="heading"
      >
        's Recipes
      </h2>
      <CreateRecipe
        handleViewRecipe={[Function]}
        match={
          Object {
            "params": Object {
              "match": Object {
                "params": Object {
                  "id": 1,
                },
              },
            },
          }
        }
      />
    </div>
    <form
      class="form-inline my-2 my-lg-0"
      onSubmit={[Function]}
    >
      <input
        aria-label="Search"
        class="form-control mr-sm-2 search"
        name="q"
        onChange={[Function]}
        placeholder="Search"
        type="text"
        value=""
      />
      <button
        className="btn my-2 my-sm-0"
        type="submit"
      >
        <i
          class="fas fa-search"
        />
        Search
      </button>
    </form>
    <hr />
  </div>
  <div
    className=""
  />
  <Pagination
    handlePageChange={[Function]}
    per_page={6}
    total=""
  />
</div>
`;
